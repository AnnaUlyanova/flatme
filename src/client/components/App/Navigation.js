import React from 'react'
import { Navbar, Nav, NavItem, Modal, Button } from 'react-bootstrap'

export default React.createClass({
  getInitialState () {
    return {show: false}
  },

  open () {
    this.setState({show: true})
  },

  close () {
    this.setState({show: false})
    this.props.login()
  },

  render () {
    const user = this.props.account.user
    const loggedInButtons = (
      <Nav pullRight bsStyle='pills'>
        <NavItem eventKey={1} href='/user/{user.id}'>{user.firstName}</NavItem>
        <NavItem eventKey={2} onClick={() => this.props.logout}>Log out</NavItem>
      </Nav>
    )
    const loggedOutButtons = (
      <Nav pullRight bsStyle='pills'>
        <NavItem eventKey={1} href='/join'>Sign up</NavItem>
        <NavItem eventKey={2} onClick={this.open}>Log in</NavItem>
      </Nav>
    )
    return (
      <div>
        <Navbar inverse collapseOnSelect>
          <Navbar.Header>
            <Navbar.Brand>
              <a href='/'>GoFlat</a>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            {user ? loggedInButtons : loggedOutButtons}
          </Navbar.Collapse>
        </Navbar>

        <Modal show={this.state.show} onHide={this.close}>
          <Modal.Header closeButton>
            <Modal.Title>Log in</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <h4>Enter your log in credentials</h4>
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={this.close}>Log in</Button>
          </Modal.Footer>
        </Modal>
      </div>
    )
  }
})
