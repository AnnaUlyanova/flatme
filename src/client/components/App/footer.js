import React from 'react'

const Footer = props => {
  return (
    <div>
      <nav className='navbar navbar-inverse navbar-fixed-bottom'>
        <div className='container-fluid'>
          <a href='https://github.com/kowhai-2016/flatme'>
            <img className='git-logo center-block' src='/images/GitHub-Mark-light-32px.png' />
          </a>
        </div>
      </nav>
    </div>
  )
}

export default Footer
