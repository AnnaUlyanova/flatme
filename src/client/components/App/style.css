.Footer .git-logo{
  margin-top: 8px;
  width:20px;
}

.Footer .signature h6{
  margin-top: 17px;
  text-align: center;
  color: lightgrey;
  font-size: 10px;
  margin-bottom: -25px;
  margin-left: -15px;
}
