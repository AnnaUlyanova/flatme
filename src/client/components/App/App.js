import React from 'react'

import Navigation from './NavContainer'

const App = props => {
  return (
    <div>
      <Navigation />
      {props.children}
    </div>
  )
}

export default App
