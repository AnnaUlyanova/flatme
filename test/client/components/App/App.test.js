/* eslint-env jest */

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

import App from '../../../../src/client/components/App'

test('App renders correctly', () => {
  const wrapper = shallow(<App />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
