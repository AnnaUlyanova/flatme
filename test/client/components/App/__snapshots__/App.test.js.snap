exports[`test App renders correctly 1`] = `
<div>
  <Connect(Navigation) />
</div>
`;
